File: leetcode/tree/balanced_binary_tree.py
def isBalanced(self, root: Optional[TreeNode]) -> bool:
    return self.f(root) != -1

def f(self, root):
    if not root:
        return 0
    
    l = self.f(root.left)
    r = self.f(root.right)
    
    if l == -1 or r == -1:
        return -1
    if abs(l - r) > 1:
        return -1
    return 1 + max(l, r)

File: leetcode/tree/balanced_binary_tree.cpp
bool isBalanced(TreeNode* root) {
	return f(root) != -1;
}

int f(TreeNode* root) {
	if (!root)
		return 0;
	int r = f(root->right);
	int l = f(root->left);
	
	if (r == -1 || l == -1) 
		return -1;
	if (abs(r - l) > 1)
		return -1;
	return max(r, l) + 1;
}
